projects/components/src/table/renderers/data-cell/table-data-cell-renderer.component.ts

```diff
@@ -1,54 +1,95 @@
 import { TableCellAlignmentType } from '../table-cell-alignment-type';
 import { TableCellRendererConstructor } from '../table-cell-renderer';
+import { TableCellRendererComponent } from '../table-cell-renderer.component';
 import { TableCellRendererService } from '../table-cell-renderer.service';
 
 @Component({
   selector: 'htc-table-data-cell-renderer',
   styleUrls: ['./table-data-cell-renderer.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="vertical-alignment-container">
-      <div class="horizontal-alignment-container" [ngClass]="this.alignment">
+    <div class="table-data-cell-renderer" [class.selected]="this.popoverOpen">
+      <htc-filter-button
+        class="filter-button"
+        *ngIf="this.columnConfig?.filterable && this.leftAlignFilterButton"
+        [metadata]="this.metadata"
+        [attribute]="this.columnConfig?.attribute"
+        [value]="this.componentRef?.instance.value"
+        (popoverOpen)="this.popoverOpen = $event"
+      ></htc-filter-button>
+      <div class="cell-renderer-content" [ngClass]="this.alignment">
         <ng-container *ngComponentOutlet="this.rendererConstructor; injector: this.rendererInjector"></ng-container>
       </div>
+      <htc-filter-button
+        class="filter-button"
+        *ngIf="this.columnConfig?.filterable && !this.leftAlignFilterButton"
+        [metadata]="this.metadata"
+        [attribute]="this.columnConfig?.attribute"
+        [value]="this.componentRef?.instance.value"
+        (popoverOpen)="this.popoverOpen = $event"
+      ></htc-filter-button>
     </div>
   `
 })
 export class TableDataCellRendererComponent implements OnInit {
+  @Input()
+  public metadata?: FilterAttribute[];
+
   @Input()
   public columnConfig?: TableColumnConfig;
 
   @Input()
   public index?: number;
 
   @Input()
-  public cellData?: unknown;
+  public rowData?: TableRow;
 
   @Input()
-  public rowData?: Dictionary<unknown>;
+  public cellData?: unknown;
 
   public alignment?: TableCellAlignmentType;
+  public leftAlignFilterButton: boolean = false;
+  public popoverOpen: boolean = false;
 
   public rendererConstructor?: TableCellRendererConstructor;
   public rendererInjector?: Injector;
+  public componentRef?: ComponentRef<TableCellRendererComponent<unknown, unknown>>;
 
   public constructor(
     private readonly injector: Injector,
-    private readonly tableCellRendererService: TableCellRendererService
+    private readonly tableCellRendererService: TableCellRendererService,
+    private readonly componentFactoryResolver: ComponentFactoryResolver
   ) {}
 
   public ngOnInit(): void {
     if (this.columnConfig === undefined) {
       throw new Error('Table column config undefined');
     }
 
+    if (this.columnConfig.filterable && !this.columnConfig.attribute) {
+      throw new Error(`Table attribute required to enable filtering for field '${this.columnConfig.field}'`);
```

Comment:
this goes to my other comment about the api - no need to have both configs.

Reply:
Removing filterable.